src/noit_metric_director.c

```diff
@@ -436,6 +437,13 @@ handle_metric_buffer(const char *payload, int payload_len,
 
         int rv = noit_message_decoder_parse_line(message, has_noit);
 
+        uint64_t drop_before_ms;
+        if((drop_before_ms = ck_pr_load_64(&drop_before_threshold_ms)) > 0) {
+          if(message->value.whence_ms < drop_before_ms) {
+            goto bail;
+          }
+        }
+
```

Comment:
Comments:
1) We could have done the check before parsing the message using `get_message_time`. This would have added additional parsing overhead for the common case (?) that the message arrives in time. Here we just have to pay for a load + compare. On the flip-side the case of late arriving data involves calloc+free of a metric_message_t and parsing of the full message. Not ideal, but much better than forwarding that stuff all the way to lua.

2) Not sure the ck_pr_load_64 is necessary here. Maybe we could have just worked with the global variable instead of making a stack copy?! At least in my use case the drop_before_threshold only ever increases. So stale reads do cause a few messages more to pass through.

Reply:
Ck stuff is unnecessary here.